library/packages/src/lib/y2packager/product_upgrade.rb

```diff
@@ -94,13 +105,43 @@ def new_base_product
       # @return [<String>] Product names which obsoletes this product.
       def will_be_obsoleted_by(old_product_name)
         installed = Y2Packager::Product.installed_products.map(&:name)
-        MAPPING.each_with_object([]) do |(products, obsoleted_by), a|
-          if products.include?(old_product_name) && (products - installed).empty?
-            a << obsoleted_by
+        selected_products = Y2Packager::Product.with_status(:selected).map(&:name)
+
+        old_products = MAPPING.keys.sort_by(&:size).reverse.find do |products|
```

Comment:
I think this tricky line deserve a comment explaining why ( e.g. why sort by size. I have to check how constant look to understand it ). and also lines below is not easy to understand as complexity is quite high.